CMakeLists.txt

```diff
@@ -10,8 +10,19 @@ endif()
 
 find_package(Threads)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# if dmclock is included by `add_subdirectory()`, check parent project's settings
+if(PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  # standalone
+  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+else()
+  # embedded
+  if(CMAKE_CXX_STANDARD OR
+      CMAKE_CXX_FLAGS MATCHES "-std=c\\+\\+" OR
+      CMAKE_CXX_FLAGS MATCHES "-std=gnu\\+\\+")
+    # use parent project's settings
+  endif()
```

Comment:
shouldn't there be an else() case that does set CMAKE_CXX_STANDARD? maybe flip it around:
```cmake
# use parent project's settings if available
if(NOT CMAKE_CXX_STANDARD AND NOT CMAKE_CXX_FLAGS MATCHES "-std=")
  set(CMAKE_CXX_STANDARD 11)
  set(CMAKE_CXX_STANDARD_REQUIRED ON)
endif()
```

Reply:
@cbodley yeah, i removed the check for standalone/embedded modes. as, in theory, use can also pass `CMAKE_CXX_FLAGS` and/or `CMAKE_CXX_STANDARD` by `cmake` command line. 

also, i still think

- it's easier to understand to avoid `NOT` operator
- `-std=` is not good enough. what if user passes options like `-std=gnu99`. it's probably pedantic. but `-std=c++17` and `-std=gnu++17` are more self-explained than `-std=`, IMHO.

also i consolidate these two regex into a single one, hopefully it's better.